server/controllers/finance/reports/cashflow/index.js

```diff
@@ -113,14 +113,16 @@ function queryIncomeExpense (params, dateFrom, dateTo) {
             pj.account_id, pj.entity_uuid, pj.currency_id, pj.trans_id,
             pj.description, pj.comment, pj.origin_id, pj.user_id
           FROM posting_journal pj
-          WHERE pj.account_id IN (?) AND DATE(pj.trans_date) >= DATE(?) AND DATE(pj.trans_date) <= DATE(?)
+          WHERE pj.account_id IN (?) AND DATE(pj.trans_date) >= DATE(?) AND DATE(pj.trans_date) <= DATE(?) AND pj.origin_id <> 10 
```

Comment:
What is the `origin_id`?  Is this simply the `type_id` of the voucher?

If so, you don't need to do `AND pj.record_uuid NOT IN (SELECT reference_uuid FROM voucher WHERE type_id = 10)`.  That will be an expensive query!